apps/src/templates/SchoolInfoConfirmationDialog.jsx

```diff
@@ -21,24 +21,46 @@ class SchoolInfoConfirmationDialog extends Component {
 
   constructor(props) {
     super(props);
-
     this.state = {
       showSchoolInterstitial: false,
       schoolName: props.schoolName,
       isOpen: true
     };
   }
 
+  static defaultProps = {
+    onUpdate: () => {},
+    schoolName: ''
+  };
+
+  componentDidMount() {
+    const {schoolName} = this.state;
+    if (!schoolName && schoolName.length > 0) {
+      fetch('/dashboardapi/v1/users/me/school_name')
```

Comment:
Shouldn't this be supplied by the controller?

Reply:
Yes, it is supplied by the controller.  I added a controller method to get the school name

Reply:
Sorry if I've just forgotten a past PR but how do we decide when to show this dialog again? Where is that code?

Reply:
Good question, I sent another PR that includes the code.  I added a helper method that contains the logic for when to show the dialog.